lib/nexaas/async/collector/result.rb

```diff
@@ -11,19 +11,35 @@ module Async
     module Collector
       class Result
 
-        attr_reader :scope_id, :id, :content
+        attr_reader :scope_id, :id, :object
 
         def initialize(scope_id, id)
           @scope_id = scope_id
           @id = id
         end
 
         def content
-          _content['content'] if content_is_ready?
+          object['content'] if content_is_ready?
         end
 
         def content_is_ready?
-          _content && _content['scope_id'] == scope_id && _content['content']
+          object && object['scope_id'] == scope_id && object['content']
+        end
+
+        def is_file?
+          object && !object['file'].nil?
+        end
+
+        def filename
+          object && object['file'] && object['file']['filename']
```

Comment:
As we support ruby 2.3+, could use the [dig](http://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig) method is this cases